### DISCUSSION TITLE
Where do the serverless functions come from on vercel even not defined any?

### DISCUSSION THREAD

**Discussion by ttiras** [vercel/next.js#49451] [Help]
### Summary
Rookie question here, i have an app on next.js 13.3.3 appDir enabled. 

There is a serverless function for all the routes in my app and i don't know where they come from.

For example i have a page "/blog" and when it is requested a log appears with status: 200 and type: request, but also a "type: Serverless function", "Function: /blog.rsc" log appears with error too and this happens with all the routes.
I would appreciate if somebody could help me with this.

### Additional information

![Screenshot 2023-05-08 at 12 33 36](https://user-images.githubusercontent.com/26274687/236818436-2073dbaa-1cb8-40d3-8443-975a4742790e.png)

**Comment by diegohaz** [comment]
Yeah, [I'm also trying to understand what's happening there](https://twitter.com/diegohaz/status/1655566842360897536).

It looks like Vercel generates ISR functions for all pages in the app dir by default and occasionally tries to revalidate them at runtime. But, since [the runtime environment is different from the build environment](https://twitter.com/diegohaz/status/1655277266736218113), you may get some errors and even end up with the page regenerated in an invalid state.

I tried setting `dynamic` to `error`, `force-static`, and everything. But couldn't solve this.
